src/subcommands/analyze.rs

```diff
@@ -23,7 +23,7 @@ pub struct AnalyzeCommand {
 
     /// Number of parallel jobs to run. Defaults to the number of CPUs.
     #[clap(short = 'j', long = "jobs", value_parser)]
-    pub parallel_build_jobs: Option<u32>,
```

Comment:
why i32?

Reply:
i32 is the type used by cargo. if we use u32 here, we get
```rust
error[E0308]: mismatched types
   --> src/cargo_project.rs:159:13
    |
157 |         compile_options.build_config = BuildConfig::new(
    |                                        ---------------- arguments to this function are incorrect
158 |             &config,
159 |             self.config.parallel_build_jobs,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `u32`
    |
    = note: expected enum `std::option::Option<i32>`
               found enum `std::option::Option<u32>`
note: associated function defined here
   --> /home/jayvdb/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-0.68.0/src/cargo/core/compiler/build_config.rs:64:12
    |
64  |     pub fn new(
    |            ^^^

```

Reply:
The reason it can be negative is hinted at in the code https://github.com/rust-lang/cargo/blob/master/src/cargo/core/compiler/build_config.rs#L84

Explanation in the docs at
https://doc.rust-lang.org/cargo/commands/cargo-bench.html?highlight=jobs#miscellaneous-options

Reply:
Ah nice! Cool